### PR TITLE
Support `struct` in `struct`

### PR DESCRIPTION
**Pull request by YusukeHosonuma**
Support `struct` in `struct` such as `Dog` has `Owner`. (and some refactoring)

### CODE REVIEW

File: Sources/Function.swift [hunk 1]
```diff
@@ -3,25 +3,38 @@
 func elementString<T>(_ x: T, debug: Bool, pretty: Bool) -> String {
     let mirror = Mirror(reflecting: x)
 
-    let typeName = type(of: x)
+    let typeName = String(describing: mirror.subjectType)
```
Comment (YusukeHosonuma) [inline Sources/Function.swift:6]: Bugfix: not retrieve really type name when `T` is `Any` type.

File: Sources/Function.swift [hunk 2]
```diff
@@ -70,14 +83,8 @@ func prettyDictionaryString<K, V>(_ d: [K: V], debug: Bool) -> String {
...
 
-    if mirror.children.count == 1, !debug {
-        guard let value = mirror.children.first?.value else { preconditionFailure() }
-        if let string = value as? String {
-            return "\"\(string)\""
-        } else {
-            return String(describing: value)
-        }
-    }
+    // Note: this function support Optional type currently that inclue 1 children.
```
Comment (sahara-ooga) [inline Sources/Function.swift]:
```suggestion
    // Note: this function currently supports Optional type that includes a child.
```

File: Sources/Function.swift [hunk 3]
```diff
@@ -3,25 +3,38 @@
...
-            "\($0.label ?? "-"): " + valueString($0.value, debug: debug)
-        }
+    }
 
-        if pretty, fields.count > 1 {
-            let tailFields = fields.dropFirst()
-                .map { $0.indent(size: prefix.count) }
-                .joined(separator: ",\n")
+    // Optional<T>
+    if typeName.starts(with: "Optional<") { // TODO: better judge is exist?
```
Comment (sahara-ooga) [inline Sources/Function.swift:14]: > // TODO: better judge is exist?

As you've mentioned in slack, we can use  `displayStyle` property of Mirror type.
https://developer.apple.com/documentation/swift/mirror/displaystyle

```suggestion
    if case .optional = mirror.displayStyle
```

Comment (sahara-ooga) [reply Sources/Function.swift:14]: The suggestion is not confirmed yet.

File: Sources/Function.swift (hunk 2 above)
Comment (sahara-ooga) [reply Sources/Function.swift]: I wonder what is returned as `displayStyle` if `x` is `Any` type. `Class` or `Struct`?

File: Sources/Function.swift (hunk 3 above)
Comment (YusukeHosonuma) [reply Sources/Function.swift:14]: Thanks for comment, I'll change it in another PR :-)

File: Sources/Function.swift (hunk 2 above)
Comment (YusukeHosonuma) [reply Sources/Function.swift]: This function `valueString()` is assume limited argument that has no fields (`Mirror#children) such as `String` or `Int` and other Basic type in Swift. This function also supports `Optional` type, but `Optional` type has one field (`Mirror#children.count == 1`). This comment is explains it.

Note:
I'll write test for this function at codes are organized.

**Review by po-miyasaka — APPROVED** [review]
🙆‍♀️🙆‍♂️

**Pull request merged**